algorithms/src/main/kotlin/io/kinference/algorithms/gec/preprocessing/Vocabulary.kt

```diff
@@ -0,0 +1,109 @@
+package io.kinference.algorithms.gec.preprocessing
+
+import java.io.File
+import java.io.InputStream
+
+/**
+ * Vocabulary for tokens/tags
+ */
+class Vocabulary {
```

Comment:
`GECVocabulary`? And also, let's make it a data class with `token2index` and `index2token` in costructor -- in that case you'll be able to fully remove generated functions for hashCode and equals.